source/extensions/filters/http/common/compressor/compressor.cc

```diff
@@ -140,20 +224,15 @@ Http::FilterHeadersStatus CompressorFilter::encodeHeaders(Http::ResponseHeaderMa
 }
 
 Http::FilterDataStatus CompressorFilter::encodeData(Buffer::Instance& data, bool end_stream) {
-  if (!skip_compression_) {
-    config_->stats().total_uncompressed_bytes_.add(data.length());
-    compressor_->compress(data, end_stream ? Envoy::Compression::Compressor::State::Finish
-                                           : Envoy::Compression::Compressor::State::Flush);
-    config_->stats().total_compressed_bytes_.add(data.length());
-  }
-  return Http::FilterDataStatus::Continue;
+  return maybeCompress(response_compressor_, config_->responseDirectionConfig().stats(), data,
+                       end_stream);
 }
 
 Http::FilterTrailersStatus CompressorFilter::encodeTrailers(Http::ResponseTrailerMap&) {
-  if (!skip_compression_) {
+  if (response_compressor_ != nullptr) {
     Buffer::OwnedImpl empty_buffer;
-    compressor_->compress(empty_buffer, Envoy::Compression::Compressor::State::Finish);
-    config_->stats().total_compressed_bytes_.add(empty_buffer.length());
+    response_compressor_->compress(empty_buffer, Envoy::Compression::Compressor::State::Finish);
```

Comment:
I believe this is no longer needed? maybeCompress is finishing the compression now.

Reply:
This code was added to fix #6894.